packages/amplify_analytics_pinpoint/test/amplify_analytics_pinpoint_test.dart

```diff
@@ -26,95 +26,235 @@ void main() {
 
   TestWidgetsFlutterBinding.ensureInitialized();
 
-  setUp(() {
-    analyticsChannel.setMockMethodCallHandler((MethodCall methodCall) async {
-      return true;
-    });
-  });
+  final eventName = 'test';
+  final userId = 'testUser';
+  final name = 'Darth Vader';
+  final email = 'vader@sith.com';
+  final plan = 'plan';
 
   tearDown(() {
     analyticsChannel.setMockMethodCallHandler(null);
   });
 
   // test sending basic events
-  test('recordEvent results in true', () async {
-    var event = AnalyticsEvent("test");
+  test('recordEvent correctly serializes input to method channel', () async {
+    analyticsChannel.setMockMethodCallHandler((MethodCall methodCall) async {
+      expect(methodCall.arguments,
+          {'name': eventName, 'propertiesMap': {}, 'propertiesTypesMap': {}});
+    });
 
-    analytics.recordEvent(event: event);
+    var event = AnalyticsEvent(eventName);
+    expect(analytics.recordEvent(event: event), completes);
   });
 
-  test('recordEvent results in true', () async {
-    var event = AnalyticsEvent("test");
+  test('flushEvents executes successfully', () async {
+    analyticsChannel.setMockMethodCallHandler((MethodCall methodCall) async {});
+    expect(analytics.flushEvents(), completes);
+  });
 
-    event.properties.addBoolProperty("boolKey", true);
-    event.properties.addDoubleProperty("doubleKey", 10.0);
-    event.properties.addIntProperty("intKey", 10);
-    event.properties.addStringProperty("stringKey", "stringValue");
+  test('enable executes successfully', () async {
+    analyticsChannel.setMockMethodCallHandler((MethodCall methodCall) async {});
+    expect(analytics.enable(), completes);
+  });
 
-    analytics.recordEvent(event: event);
+  test('disable executes successfully', () async {
+    analyticsChannel.setMockMethodCallHandler((MethodCall methodCall) async {});
+    expect(analytics.disable(), completes);
   });
 
-  test('registerGlobalProperties results in true', () async {
-    var globalProperties = AnalyticsProperties();
+  test(
+      'unregisterGlobalProperties correctly serializes input to method channel',
+      () async {
+    final propertiesToUnregister = ['name'];
+    analyticsChannel.setMockMethodCallHandler((MethodCall methodCall) async {
+      expect(methodCall.arguments, propertiesToUnregister);
+    });
+    expect(
+        analytics.unregisterGlobalProperties(
+            propertyNames: propertiesToUnregister),
+        completes);
```

Comment:
Nit: shouldn't this be indented so that lines 74 and 75 are aligned?

Reply:
disclosure: not being acutely aware of dart indenting, I have been letting the vscode formatter handle this on save automatically and trusting it.

I thought your comment made sense, but when I tried to make that change, the formatter kept making it like this. I **think** this is correct now bc lines 73 and 75 are both arguments to `expect` so they are aligned. Line 74 is indented bc it's just an extension of line 73 that is too long to fit. Does that make sense?

Reply:
good call, thanks for taking a look into that